src/components/taskDetails/taskDependency/index.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { ChangeEvent, FC, useState } from 'react';
+import { Textarea } from '..';
+import { TaskDependencyProps } from '@/interfaces/taskDetails.type';
+import DependencyList from '@/components/taskDetails/taskDependency/DependencyList';
+import { parseDependencyValue } from '@/utils/parseDependency';
+
+const TaskDependency: FC<TaskDependencyProps> = ({
+    taskDependencyIds,
+    isEditing,
+    updatedDependencies,
+    handleChange,
+}) => {
+    const [editedDependencies, setEditedDependencies] =
+        useState<string[]>(updatedDependencies);
+
+    const handleDependenciesChange = (
+        event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+    ) => {
+        const { value } = event.target;
+        const updatedDependencies = parseDependencyValue(value);
+        setEditedDependencies(updatedDependencies);
+        handleChange(event);
+    };
+
+    return (
+        <>
+            {isEditing && (
```

Comment:
Are we writing tests when `isEditing` is true and when `isEditing` is false both?

Reply:
yes

Reply:
yes

Reply:
Can you please tell in which file this in?

Reply:
If isEditing is false then it shows depdencyList component which i already mention

Reply:
But are there tests for this?

Reply:
![image](https://github.com/Real-Dev-Squad/website-status/assets/111434418/f1ce607e-ecce-458f-ba31-9adde0774d91)